tests/launch.rs

```diff
@@ -1,24 +1,227 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use kvm_ioctls::Kvm;
+use vmm_sys_util::*;
 
 use tdx::launch::{TdxVcpu, TdxVm};
 use tdx::tdvf;
 
+// `mov eax,1000h` will set the value in the register eax (and rax since they both share the bottom 32 bits) to 1000h
+// `jmp *%rax` will jump the program to the address that rax contains, which in this case will be 1000h
+const FIRMWARE: &[u8; 7] = &[
+    0xb8, 0x00, 0x10, 0x00, 0x00, // mov eax, 1000h
+    0xff, 0xe0, // jmp *%rax
+];
+
 #[test]
 fn launch() {
-    let mut kvm_fd = Kvm::new().unwrap();
+    const KVM_CAP_GUEST_MEMFD: u32 = 234;
+    const KVM_CAP_MEMORY_MAPPING: u32 = 236;
 
     // create vm
+    let mut kvm_fd = Kvm::new().unwrap();
     let tdx_vm = TdxVm::new(&kvm_fd, 100).unwrap();
-    let caps = tdx_vm.get_capabilities().unwrap();
-    let _ = tdx_vm.init_vm(&kvm_fd, &caps).unwrap();
+    let _caps = tdx_vm.get_capabilities().unwrap();
+    let _ = tdx_vm.init_vm(&kvm_fd).unwrap();
+
+    // get tdvf sections
+    let mut firmware = std::fs::File::open("/usr/share/edk2/ovmf/OVMF.inteltdx.fd").unwrap();
```

Comment:
This is a local file on your system. Add it to `tests/data/OVMF.inteltdx.fd` and refer to it there.

Reply:
Fixed. Just one comment on the above